modules/cardano-faucet.nix

```diff
@@ -0,0 +1,55 @@
+with (import ./../lib.nix);
+
+globals: params:
+{ config, pkgs, lib, ...}:
+
+let
+  recaptcha = {
+    siteKey = lib.fileContents ../static/recaptcha_site_key;
+    secretKey = lib.fileContents ../static/recaptcha_secret_key;
+  };
+  explorerURL = "http://cardano-explorer.cardano-testnet.iohkdev.io/";
+  faucetFrontend = pkgs.runCommand "faucet-frontend" {} ''
+    mkdir -p $out
+    sed -e 's|recaptchaSiteKey:.*|recaptchaSiteKey: "${recaptcha.siteKey}",|g' \
```

Comment:
I don't like that we're reading the recaptcha key from a private static area, but then injecting it plain text into the store. Essentially, if the server is hacked under any user, this makes the key visible. Couldn't we deploy this as a deploy key and have the faucet take a path to the file containing the key so it doesn't end up in the nix store? Since we own the faucet code, I don't see why we wouldn't want to go this route.

Reply:
True. The recaptcha site key is public and injected into HTML. The recaptcha secret key is private and shouldn't be in the nix store. I already asked if faucet code could read the recaptcha secret key from a file but everything has to be asked for at least twice.